server/utils/programmeUtils.test.ts

```diff
@@ -0,0 +1,31 @@
+import { programmeFactory, programmePrerequisiteFactory } from '../testutils/factories'
+import { prerequisiteSummaryListRows, programmeListItems } from './programmeUtils'
+
+describe('programmeUtils', () => {
+  describe('prerequisiteSummaryListRows', () => {
+    it('converts prerequisites into summary list rows', () => {
+      const prerequisites = programmePrerequisiteFactory.buildList(3)
+
+      const expected = prerequisites.map(prerequisite => {
+        return { key: { text: prerequisite.key }, value: { text: prerequisite.value } }
+      })
+
+      expect(prerequisiteSummaryListRows(prerequisites)).toEqual(expected)
+    })
+  })
+
+  describe('programmeListItems', () => {
+    it('converts programmes into programme list items (converting the prerequisites)', () => {
+      const programmes = programmeFactory.buildList(3)
+
+      const expected = programmes.map(programme => {
+        return {
+          ...programme,
+          programmePrerequisites: prerequisiteSummaryListRows(programme.programmePrerequisites),
+        }
+      })
```

Comment:
Yeah, I think it would be good to write this all out - we'll only need to do it once as this is a util that'll be reused, but I think it would also help the tests serve as standalone documentation of the code, rather than tests of functionality.

Reply:
I've updated this - what do you think about the `programmeListItems` format? It's slightly more explicit, but I still kept the spreading and the use of `prerequisiteSummaryListRows`. The spreading indicates "it returns a value that has all the key/value pairs of the original programmes", then the use of `prerequisiteSummaryListRows` indicates "except that the value of `programmePrerequisites` is overwritten with the output of this method". Since that method is already tested above, it feels like we shouldn't be testing it's output here too, only that this method overwrites `programmePrerequisites` with its output?

Reply:
Yeah I think this is fine - it's pretty clear from reading the test what's going on now, thanks for updating.